### PR TITLE
fix(upgrades): activate new plugins on system upgrade

### LINKED ISSUE #10603: activity plugin should be enabled after upgrade

### LINKED ISSUE #11699: should system_log plugin be activated by default
if you dont want it, you can turn it off... if you turn it on later, you are missing history that you thought you had

### LINKED ISSUE #11891: ``Batch`` is not an intuitive class name
I have to look up what the class name is that I should extend for an Async Upgrade. Maybe add an alias a-la ``ElggAsyncUpgrade``

### LINKED ISSUE #11915: Exceptions in async upgrades
Currently if an exception occurs, the upgrade run indefinitely with 0 success and failures

### PR DESCRIPTION
**Pull request by hypeJunction**
Activates plugins added in 3.0 during upgrade.
Simplifies some of the upgrade logic.

Fixes #10603
Fixes #11915
Fixes #11891
Fixes #11699

### CODE REVIEW

File: engine/classes/Elgg/UpgradeService.php [hunk 1]
```diff
@@ -108,33 +110,42 @@ public function run($async = false) {
...
-				$this->system_messages->addSuccessMessage($msg);
+		foreach ($upgrades as $key => $upgrade) {
+			if (!$upgrade->isAsynchronous()) {
+				$execute($upgrade);
+			}
+		}
 
-				$this->logger->log("Finished upgrade {$upgrade_name}", Logger::NOTICE);
+		if ($async) {
+			foreach ($upgrades as $upgrade) {
```
Comment (hypeJunction) [inline engine/classes/Elgg/UpgradeService.php]: grab pending again

File: engine/classes/Elgg/UpgradeService.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
 use Elgg\Upgrade\Batch;
 use Elgg\Upgrade\Locator;
 use Elgg\Upgrade\Result;
+use Elgg\Upgrades\ActivateNewPlugins;
```
Comment (hypeJunction) [inline engine/classes/Elgg/UpgradeService.php]: not used

File: views/default/admin/upgrades.php [hunk 3]
```diff
@@ -1,29 +1,13 @@
-<?php
+-<?php
```
Comment (jeabakker) [inline views/default/admin/upgrades.php]: what happened here??

Comment (hypeJunction) [reply views/default/admin/upgrades.php]: Meh. Don't know. Fingers faster than my brain.

Comment (jeabakker) [reply views/default/admin/upgrades.php]: > Fingers faster than my brain

that can be very dangerous when programming, maybe not in other parts of your life though ;)

Comment (hypeJunction) [reply views/default/admin/upgrades.php]: could do a manual test from 2.3?

**Comment by jdalsem** [comment]
will do a big manual test today

**Pull request merged**